File: src/Factor/Factor.java
package Factor;
import Exception.RuntimeErrorException;
import FormulaParser.Expression;


public abstract class Factor implements Expression {

	private Expression child;
	
	public Expression getChild() {
		return this.child;
	}
	
	public void setChild(Expression arg0) {
		this.child = arg0;
	}
	
	@Override
	abstract public String evaluate() throws RuntimeErrorException;

}

File: src/main/kotlin/com/lucciola/factor/Factor.kt
package com.lucciola.factor

import com.lucciola.exception.RuntimeErrorException
import com.lucciola.formulaparser.Expression
import com.lucciola.termination.Termination

abstract class Factor protected constructor(val child: Termination) : Expression {

    @Throws(RuntimeErrorException::class)
    abstract override fun evaluate(): String

}
